src/components/Card.js

```diff
@@ -15,13 +27,32 @@ export default class Card {
   }
 
   _handleLike = () => {
-    this._likeButton.classList.toggle("card__like-button_active");
+    if (this.isLiked) {
+      this._unlikeHandler(this);
+      this.isLiked = false;
```

Comment:
` this.isLiked = false;` or `this.isLiked = true;` should be set only in `then`

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.

Reply:
Moved to the add and remove like functions that are used in the then statements